tools/devops/automation/templates/main-stage.yml

```diff
@@ -163,12 +163,26 @@ parameters:
   type: string
   default: ''
 
+- name: dependsOn
+  type: string
+  default: ''
+
+- name: dependsOnResult
+  type: string
+  default: ''
+
 stages:
 
 - ${{ if eq(parameters.runGovernanceTests, true) }}:
   - stage: governance_checks
     displayName: '${{ parameters.stageDisplayNamePrefix }}Governance Checks'
-    dependsOn: []
```

Comment:
Is it allowed to depend on an empty string?


```
dependsOn: ''
```

if so, you could just remove the condition:

```
dependsOn: ${{ parameters.dependsOn }}
```

Alternatively you can change the default to to be  `[]`, in which case you could also get rid of the condition and simplify the code.

Reply:
we will have to test to be honest, yaml has its things